File: src/main/java/models/p/dto/RequestForPatentRecognitionDTO.java
package models.p.dto;

import models.p.*;

import javax.xml.bind.annotation.XmlAttribute;
import javax.xml.bind.annotation.XmlElement;

public class RequestForPatentRecognitionDTO {

    @XmlElement(name = "Information_for_institution")
    public InformationForInstitutionDTO informationForInstitution;
    @XmlElement(name = "Patent_name")
    public PatentNameDTO patentName;
    @XmlElement(name = "Applicant")
    public TPersonDTO applicant;
    @XmlElement(name = "Inventor")
    public InventorDTO inventor;
    @XmlElement(name = "Proxy")
    public ProxyDTO proxy;
    @XmlElement(name = "Delivery_address")
    public DeliveryAddressDTO deliveryAddress;
    @XmlElement(name = "Delivery_type")
    public DeliveryTypeDTO deliveryType;
    @XmlElement(name = "Application_information")
    public ApplicationInformationDTO applicationInformation;
    @XmlElement(name = "Priority_rights_recognition_from_earlier_applications")
    public PriorityRightsRecognitionFromEarlierApplicationsDTO priorityRightsRecognitionFromEarlierApplications;
    @XmlAttribute(name = "address")
    public String address;
    @XmlAttribute(name = "institution")
    public String institution;
}

File: frontend/src/app/dto/RequestForPatentRecognitionDTO.ts
import {Person, Individual, Address} from "./User";

export interface RequestForPatentRecognitionDTO {

  informationForInstitution: InformationForInstitution;
  patentName: PatentName;
  applicant: Person;
  inventor: Inventor;
  proxy: Proxy;
  deliveryAddress: DeliveryAddress;
  deliveryType: DeliveryType;
  applicationInformation: ApplicationInformation;
  priorityRightsRecognitionFromEarlierApplications: PriorityRightsRecognitionFromEarlierApplications;
  address: string;
  institution: string;
  isAccepted: boolean;
}

export interface InformationForInstitution {
  applicationNumber: string;
  receiptDate: string;
  submissionDate: string;
}

export interface PatentName {
  serbianPatentName: string;
  englishPatentName: string;
}

export interface DeliveryAddress extends Address{}

export interface DeliveryType {
  electronicDelivery: boolean;
  deliveryInPaperForm: boolean;
}

export interface ApplicationInformation {
  originalApplicationNumber: string;
  originalApplicationSubmissionDate: string;
  supplementaryApplication: boolean;
  separateApplication: boolean;
}

export interface PriorityRightsRecognitionFromEarlierApplications {
  earlierApplications: EarlierApplications;
}

export interface EarlierApplications {
  earlierApplication: EarlierApplication[];
}

export interface EarlierApplication {
  earlierApplicationSubmissionDate: string;
  earlierApplicationNumber: string;
  countryOrOrganizationDesignation: string;
}

export interface Inventor extends Individual {
  doesNotWantToBeListed: boolean;
}

export interface Proxy extends Individual {
  proxyForRepresentation: boolean;
  attorneyForReceivingLetters: boolean;
}
